src/darwin/Framework/CHIP/MTRCommissioningOperation.h

```diff
@@ -0,0 +1,65 @@
+/**
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRCommissioningDelegate.h>
+#import <Matter/MTRCommissioningParameters.h>
+#import <Matter/MTRDeviceController.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+MTR_UNSTABLE_API
+@interface MTRCommissioningOperation : NSObject
+/**
+ * Initialized via initWithParameters:setupPayload:delegate:queue:
+ */
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Prepare to do a new commissioning of a device with the given parameters and
+ * the given setup payload (QR code, manual pairing code, etc).  Returns nil if
+ * the payload is not valid.
+ *
+ * The deviceAttestationDelegate property of MTRCommissioningParameters will be
+ * ignored. Device attestation notifications will be delivered to the
+ * MTRCommissioningDelegate instead.  The failSafeTimeout property of
+ * MTRCommissioningParameters will be respected.
+ *
+ * The provided delegate will be notified about various things as commissioning
+ * proceeds.  The calls into the delegate will happen on the provided queue.
+ */
+- (nullable instancetype)initWithParameters:(MTRCommissioningParameters *)parameters
+                               setupPayload:(NSString *)payload
+                                   delegate:(id<MTRCommissioningDelegate>)delegate
+                                      queue:(dispatch_queue_t)queue;
+
+/**
+ * Start commissioning with the given controller (which identifies the fabric
+ * the commissionee should be commissioned into).  The delegate will be notified
+ * if there are any failures.
+ */
+- (void)startWithController:(MTRDeviceController *)controller;
+
+/**
+ * Stop commissioning.  This will typically result in
+ * commissioning:failedWithError: callbacks to delegates.
+ */
+- (void)stop;
```

Comment:
Should this return a BOOL to indicate if we stopped? So if e.g. the call was made too late and we were already done the result will be NO.

Reply:
Done.